lib/parser.js

```diff
@@ -1,33 +1,47 @@
 const fs = require('fs');
-
-const { Parser, convertToOldAPI } = require('@asyncapi/parser/cjs');
-const { OpenAPISchemaParser } = require('@asyncapi/openapi-schema-parser');
-const { AvroSchemaParser } = require('@asyncapi/avro-schema-parser');
-const { RamlDTSchemaParser } = require('@asyncapi/raml-dt-schema-parser');
-const { ProtoBuffSchemaParser } = require('@asyncapi/protobuf-schema-parser');
+const { convertToOldAPI } = require('@asyncapi/parser');
+const { ConvertDocumentParserAPIVersion, NewParser } = require('@smoya/multi-parser');
 
 const parser = module.exports;
 
-const defaultParser = new Parser({
-  schemaParsers: [
-    OpenAPISchemaParser(),
-    AvroSchemaParser(),
-    RamlDTSchemaParser(),
-    ProtoBuffSchemaParser(),
-  ],
-});
+/**
+ * Conver the template defined value `apiVersion: 'v1'` to only contain the numeric value `1`.
+ */
+parser.sanitizeTemplateApiVersion = (apiVersion) => {
+  if (apiVersion && apiVersion.length > 1) {
+    return apiVersion.substring('1');
+  }
+  return apiVersion;
+};
 
 parser.parse = (asyncapi, oldOptions, generator) => {
+  let apiVersion = this.sanitizeTemplateApiVersion(generator.templateConfig.apiVersion);
+  // Defaulting to v1 parser to convert it to the old parserAPI afterwards.
+  if (!this.usesNewAPI(generator.templateConfig)) {
+    apiVersion = '1';
```

Comment:
why we default to `1`? kinda do not get `// Defaulting to v1 parser to convert it to the old parserAPI afterwards.`

why not `undefined`?

Reply:
If we want the old parser model, we still need to parse the document using one of the parser versions, for then to convert it afterward if necessary.

How can we clarify this better? 🤔

Reply:
oh, ok, during review I did not look into `@smoya/multi-parser` code.

> // Defaulting to v1 parser to convert it to the old parserAPI afterwards.

by `old parserAPI` you do not mean JS Parser v1 but JS Parser v2 (Parser-API v1) ? right?

so `@smoya/multi-parser` is just for stuff related to Parser-API

---
I got confused, as for me `old` is the old API before invention of Parser-API

later in code you write

> // Convert to old version

and then it is related to JS Parser v1

---
> How can we clarify this better?

so just be precise with naming and use phrases like `JS Parser v1`, `Parser-API v1`, `Parser-API v2`

in sentence `Defaulting to v1 parser to convert it to the old parserAPI afterwards.` - like I literally have no idea what `v1 parser` is for you here as `old parserAPI` for me is `JS Parser v2` not `v1`

Reply:
I can suggests edits, but first you need to confirm if I understood stuff

Reply:
Yea that makes sense 👍 

Please do make a suggestion 🙇

Reply:
done, wdyt?

Reply:
Looks good to me 👍